hail/python/hailtop/hailctl/batch/initialize.py

```diff
@@ -0,0 +1,252 @@
+
+    from .utils import already_logged_into_service, login_to_service  # pylint: disable=import-outside-toplevel
+
+    already_logged_in = await already_logged_into_service(domain)
+    if not already_logged_in:
+        typer.secho('You are not currently logged in to Hail. Redirecting you to a login screen.', fg=typer.colors.YELLOW)
+        await login_to_service(domain)
+        typer.secho('In the future, you can use `hailctl auth login` to login to Hail.', fg=typer.colors.YELLOW)
+    else:
+        domain = get_deploy_config()._domain
+        typer.secho(f'You are currently logged in to Hail at domain {domain}.')
+
+    user_info = await async_get_userinfo()
+    username = user_info['username']
+    hail_identity = user_info['hail_identity']
+    trial_bp_name = user_info['trial_bp_name']
+
+    batch_client = await BatchClient.create(trial_bp_name)
+
+    async with batch_client:
+        cloud = await batch_client.cloud()
+        supported_regions = await batch_client.supported_regions()
+
+        if cloud == 'gcp':
+            remote_tmpdir, tmpdir_region, warnings = await initialize_gcp(username, hail_identity, supported_regions, verbose)
+        else:
+            remote_tmpdir = Prompt.ask('Enter a path to an existing remote temporary directory (ex: gs://my-bucket/batch/tmp)')
+            typer.secho(f'WARNING: You will need to grant read/write access to {remote_tmpdir} for account {hail_identity}', fg=typer.colors.YELLOW)
+            warnings = False
+
+            tmpdir_region = Prompt.ask('Which region is your remote temporary directory in? (Example: eastus)')
+
+    compute_region = Prompt.ask('Which region do you want your jobs to run in?', choices=supported_regions)
+
+    if tmpdir_region != compute_region and not compute_region.startswith(tmpdir_region):
+        typer.secho(f'WARNING: remote temporary directory "{remote_tmpdir}" is not located in the selected compute region for Batch jobs "{compute_region}". Found {tmpdir_region}.',
+                    fg=typer.colors.YELLOW)
+        warnings = True
+
+    if trial_bp_name:
+        set_config(ConfigVariable.BATCH_BILLING_PROJECT, trial_bp_name)
+
+    if remote_tmpdir:
+        set_config(ConfigVariable.BATCH_REMOTE_TMPDIR, remote_tmpdir)
+
+    set_config(ConfigVariable.BATCH_REGIONS, compute_region)
+
+    set_config(ConfigVariable.BATCH_BACKEND, 'service')
+
+    query_backend = Prompt.ask('Which backend do you want to use for Hail Query?', choices=['spark', 'batch', 'local'])
+    set_config(ConfigVariable.QUERY_BACKEND, query_backend)
+
+    typer.secho('--------------------', fg=typer.colors.BLUE)
+    typer.secho('FINAL CONFIGURATION:', fg=typer.colors.BLUE)
+    typer.secho('--------------------', fg=typer.colors.BLUE)
+    list_config()
+
+    if warnings:
+        typer.secho('WARNING: Initialized Hail with warnings! The currently specified configuration will result in additional ingress and egress fees when using Hail Batch.',
```

Comment:
```suggestion
        typer.secho('WARNING: The currently specified configuration will result in additional ingress and egress fees when using Hail Batch.',
```

I don't think the first sentence is very necessary, and I'm not quite sure how to interpret it.